rewrite-java-test/src/test/java/org/openrewrite/java/cleanup/RemoveInstanceOfPatternMatchTest.java

```diff
@@ -477,6 +477,40 @@ public void test(Object obj) {
             14));
     }
 
+    @Test
+    @Disabled("Not supported")
+    void negationWithoutElse() {
```

Comment:
Detail: It isn't so much the absence of the `else` which is special about this case, it is the `if` throwing an exception. There could also be an `else` which would also be allowed to dereference the `str` pattern variable.

Reply:
I think exception throwing is not a problem for the current implementation. But variable dereference outside if/else is not supported at all. I can try to fix it in this pull request, but not sure that it will be simple with a current approach. Maybe it's better to create another pull-request.

Reply:
I agree. I would also suggest to defer the two disabled tests for a later PR.